### ISSUE TITLE
curl: (28) Operation timed out

### ISSUE THREAD

**Issue by juvenrui** [Homebrew/brew#14451]
### `brew doctor` output

```shell
curl: (28) Operation timed out after 5000 milliseconds with 2742071 out of 3444439 bytes received
curl: (28) Operation timed out after 5003 milliseconds with 2600960 out of 3444439 bytes received
......
```

### Verification

- [x] My "`brew doctor` output" above says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` twice and am still able to reproduce my issue.
- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 3.6.20-143-g10845a1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 10845a1122b916dc91d04f00a4d8b7b888d52654
Last commit: 18 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 15f0ace650a2a17971d9fee339d5c18163c7d721
Core tap last commit: 3 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BAT_THEME: Monokai Extended Bright
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: subl --new-window --wait
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit skylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.3-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### What were you trying to do (and why)?

try to:
brew services list,
brew cleanup

### What happened (include all command output)?

CLI output: 

curl: (28) Operation timed out after 5000 milliseconds with 2742071 out of 3444439 bytes received
.....

### What did you expect to happen?

Reinstall xcode command tool, but not. 

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew services list  --verbose --debug

output: 

/usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.6.20-143-g10845a1\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 12.6.3\)\ curl/7.79.1 --header Accept-Language:\ en --fail --max-time 5 --retry 3 --location --remote-time --output /Users/juven/Library/Caches/Homebrew/api/formula.json --time-cond /Users/juven/Library/Caches/Homebrew/api/formula.json --compressed --silent https://formulae.brew.sh/api/formula.json
/usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.6.20-143-g10845a1\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 12.6.3\)\ curl/7.79.1 --header Accept-Language:\ en --fail --max-time 5 --retry 3 --location --remote-time --output /Users/juven/Library/Caches/Homebrew/api/formula.json --compressed --silent https://formulae.brew.sh/api/formula.json
curl: (28) Operation timed out after 5001 milliseconds with 2113326 out of 3444439 bytes received
curl: (28) Operation timed out after 5005 milliseconds with 2181349 out of 3444439 bytes received
curl: (28) Operation timed out after 5003 milliseconds with 2868078 out of 3444439 bytes received
......
```

**Comment by juvenrui** [comment]
$ brew install wget

<img width="1007" alt="截屏2023-01-28 13 07 07" src="https://user-images.githubusercontent.com/4530373/215243254-5aabfd16-f474-4eb0-8b19-09f13c68cd19.png">

**Comment by carlocab** [comment]
Does doing
```sh
export HOMEBREW_NO_INSTALL_FROM_API=1
```
help?

**Comment by juvenrui** [comment]
> Does doing
> 
> ```shell
> export HOMEBREW_NO_INSTALL_FROM_API=1
> ```
> 
> help?

Worked, thans.

**Comment by thanieljay** [comment]
> Does doing
> 
> ```shell
> export HOMEBREW_NO_INSTALL_FROM_API=1
> ```
> 
> help?

Worked for me for this issue:
curl: (22) the requested url returned error: 404

**Comment by psimanta** [comment]
> Does doing
> 
> ```shell
> export HOMEBREW_NO_INSTALL_FROM_API=1
> ```
> 
> help?

Worked for me (Ventura 13.0), Thanks !

**Comment by saymonaraujo** [comment]
> Does doing
> 
> ```shell
> export HOMEBREW_NO_INSTALL_FROM_API=1
> ```
> 
> help?

I had the same problem and this solved for me as well.
thanks !

**Comment by shockerli** [comment]
> 

THX. macOS 12.5

**Comment by shockerli** [comment]
> Does doing
> 
> ```shell
> export HOMEBREW_NO_INSTALL_FROM_API=1
> ```
> 
> help?

THX. macOS 12.5

**Comment by ivaquero** [comment]
> Does doing
> ```sh
> export HOMEBREW_NO_INSTALL_FROM_API=1
> ```
> help?

This solves almost every command except `brew livecheck`.

**Comment by gromgit** [comment]
What's not working for you with `brew livecheck`? If you're getting inconsistent results with `brew live check` and `brew lc`, that's fixed in a different PR: https://github.com/Homebrew/brew/pull/14474. `brew update` and that should be resolved.

**Comment by ivaquero** [comment]
Let me try again, then I will report more details.

**Comment by WhyTimePass** [comment]
Thank you all !

**Comment by mangomangomango0820** [comment]
I encountered the same issue, after trying with `export HOMEBREW_NO_INSTALL_FROM_API=1` and `brew update`, everything works fine now.

macOS 11.4.

**Comment by fyeeme** [comment]
same problem, ths

**Comment by doggeddog** [comment]
This problem is really annoying, please make HOMEBREW_NO_INSTALL_FROM_API=1 the default config.

**Comment by MikeMcQuaid** [comment]
@doggeddog No. Set it on your own machine.
